ring/watcher.go

```diff
@@ -0,0 +1,54 @@
+package ring
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Watcher interface {
```

Comment:
Why do we need this public interface in the package? Can it just be removed since it's not in use?

Reply:
okok, I almost used it for something else as well, but it did not improve speed